float32 x
float32 y
float32 vx
uint8 ID
uint8 target_type